### ISSUE TITLE
Can't access VLAN interfaces

### ISSUE THREAD

**Issue by overcore456** [opnsense/core#1040]
Hello

I've a little issue while we are trying to force a network segmentation. I configured a separate physical network interface (**em2**) on my OPNsense (OPNsense 16.1.17-amd64). On this interface I configured two VLANs (**VLAN 10 for Clients and VLAN 20 for Servers**) and they appear as configurable interfaces under assignments. So far so good. I assigned and enabled them with a static IP address (**vlan 10 on em2 with 192.168.10.1/24 and vlan 20 on em2 with 192.168.20.1/24**).
So everything looks fine under the overview tab and I went on to add outbound NAT and a firewall rule.

The physical firewall interface is connected to a Cisco switch port with a Trunk Port configuration (**1 untagged (native), 10 tagged, 20 tagged**) and I added two physical client with separate Access Ports (vlan 10 and vlan 20) with the IPs 192.168.10.10 and 192.168.20.20. So from my experience those two test devices should can ping the firewall VLAN interfaces in their segment but they get a "Destination host unreachable". 

192.168.10.10 -> 192.168.10.1
192.168.20.20 -> 192.168.20.1 

I also tried this setup with Access Ports only and this worked great, but as soon I configure VLANs on the OPNsense its not working any more.

I've checked the documentation and watched several Youtube videos for VLAN setups and I must admit the setup is very easy and I couldn't find any wrong configurations so I'm contacting the community. :-)
Maybe somebody can help me.

Thanks a lot.
Marco

**Comment by fichtner** [comment]
Hi Marco,

I am thinking about your outbound NAT and firewall rule, neither should be needed for a simple ping test to work. What are your rules? Are you not using automatic outbound NAT?

Did you confirm with a packet trace that (a) echo request packets arrive on your OPNsense interface, and (b) echo reply packets leave, respectively?


Cheers,
Franco

**Comment by overcore456** [comment]
Hi Franco,

I'am using manual outbound NAT rules for Source NAT behind the WAN interface, nothing more. Additionally I have two firewall rules they allow any connections from those two networks to any network.
Regarding the packet trace, I did a package capture on both VLAN firewall interfaces for ICMP and didn't receive anything. As soon I switch to Access ports and interfaces without a VLAN everything works fine.

To make sure it is not the switch I connected a Hyper-V server with a Windows 10 VM and connected this Hyper-V on a Trunk Port with exact the same VLAN configuration as the firewall and put the Virtual Machine on VLAN 10 in the VM settings. This works also fine, so I can communicate between the Virtual Machine and the test device with a taggend trunk.

Cheers,
Marco

**Comment by fichtner** [comment]
Thanks for the quick reply! Is OPNsense running on Hyper-V or hardware?

**Comment by overcore456** [comment]
My system is running on a hardware. But I also tested this scenario on a Hyper-V system aswell, but this this doesn't work aswell (could just tested this with one VLAN because Microsoft just supports one VLAN on a vSwitch with the Hyper-V Manager, for more you the VMM :-)).

**Comment by fichtner** [comment]
Just making sure. :) Would you mind pasting an ifconfig dump?

**Comment by overcore456** [comment]
# ifconfig 
em0: flags=8843<UP,BROADCAST,RUNNING,SIMPLEX,MULTICAST> metric 0 mtu 1500
	options=88<VLAN_MTU,VLAN_HWCSUM>
	nd6 options=21<PERFORMNUD,AUTO_LINKLOCAL>
	media: Ethernet autoselect (1000baseT <full-duplex>)
	status: active
em1: flags=8843<UP,BROADCAST,RUNNING,SIMPLEX,MULTICAST> metric 0 mtu 1500
	options=40098<VLAN_MTU,VLAN_HWTAGGING,VLAN_HWCSUM,VLAN_HWTSO>
	nd6 options=21<PERFORMNUD,AUTO_LINKLOCAL>
	media: Ethernet 100baseTX <full-duplex>
	status: active
em2: flags=8c02<BROADCAST,OACTIVE,SIMPLEX,MULTICAST> metric 0 mtu 1500
	options=2088<VLAN_MTU,VLAN_HWCSUM,WOL_MAGIC>
	nd6 options=21<PERFORMNUD,AUTO_LINKLOCAL>
	media: Ethernet autoselect (1000baseT <full-duplex>)
	status: active
lo0: flags=8049<UP,LOOPBACK,RUNNING,MULTICAST> metric 0 mtu 16384
	options=600003<RXCSUM,TXCSUM,RXCSUM_IPV6,TXCSUM_IPV6>
	inet 127.0.0.1 netmask 0xff000000 
	inet6 ::1 prefixlen 128 
	inet6 fe80::1%lo0 prefixlen 64 scopeid 0x7 
	nd6 options=21<PERFORMNUD,AUTO_LINKLOCAL>
em2_vlan10: flags=8843<UP,BROADCAST,RUNNING,SIMPLEX,MULTICAST> metric 0 mtu 1500
	inet6 fe80::7ae3:b5ff:fe86:b5df%em2_vlan10 prefixlen 64 scopeid 0xa 
	inet 192.168.10.1 netmask 0xffffff00 broadcast 192.168.10.255 
	nd6 options=21<PERFORMNUD,AUTO_LINKLOCAL>
	media: Ethernet autoselect (1000baseT <full-duplex>)
	status: active
	vlan: 10 parent interface: em2
em2_vlan20: flags=8843<UP,BROADCAST,RUNNING,SIMPLEX,MULTICAST> metric 0 mtu 1500
	inet6 fe80::7ae3:b5ff:fe86:b5df%em2_vlan20 prefixlen 64 scopeid 0xb 
	inet 192.168.20.1 netmask 0xffffff00 broadcast 192.168.20.255 
	nd6 options=21<PERFORMNUD,AUTO_LINKLOCAL>
	media: Ethernet autoselect (1000baseT <full-duplex>)
	status: active
	vlan: 20 parent interface: em2

**Comment by overcore456** [comment]
I just removed some confidential information. :-)

**Comment by fichtner** [comment]
I fixed up "&lt;" and "&gt;", github tends to hide it.

This has all hardware acceleration features turned on. Have you tried to disable them? Especially VLAN hardware filtering under Interfaces: Settings.

**Comment by overcore456** [comment]
On my web interface all hardware filterings are already disabled. 
![image](https://cloud.githubusercontent.com/assets/19660884/16370993/1771a800-3c87-11e6-8de8-2801d27dbb19.png)

**Comment by overcore456** [comment]
I just notices, that my client, which is running in the old/native VLAN 1 (untagged) on interface em0 can actually ping 192.168.10.1 and 192.168.20.1.

**Comment by fichtner** [comment]
Ok, was ifconfig after a full reboot with those disabled settings? They don't take effect immediately.

It looks like tagged VLAN traffic from the outside will not reach the interface. Capturing in promisc mode on your untagged interface will likely reveal that. Pinging from the GUI using a tagged interface to a reachable client will likely produce an echo request packet and send it out, the client will reply, but the response will not reach the interface again.

To recap, from the GUI, preferably both at the same time:

(1) Promisc package capture on the untagged interface to see if there are any VLANs in there
(2) Ping from a tagged interface to its respective client

**Comment by overcore456** [comment]
I don't know... the opnsense was rebooted after we installed the new 16.1.17. I think my colleague didn't changed any settings regarding hardware filter. So is there any difference between the screenshot and the ifconfig settings that I haven't seen?

How I can see if there are VLANs in the capture? Sorry I'am not that into this.

**Comment by overcore456** [comment]
Ok here is what I did and some background infos.

Old network: 192.168.0.0/24 (native VLAN1 untagged) -> FW INT 192.168.0.1
New Client network: 192.168.10.0/24 (VLAN10 tagged) -> FW INT 192.168.10.1
New Server network: 192.168.20.0/24 (VLAN20 tagged) -> FW INT 192.168.20.1

1. Did a prom capture on the old network interface (192.168.0.1) and started a ping from my normal client which is in the old network as well (192.168.0.44) to a test server (192.168.20.5) in the new VLAN 20. 
Output:
09:45:21.159473 IP 192.168.0.44 > 192.168.20.5: ICMP echo request, id 1, seq 21501, length 40
There is no ICMP reply.

I let this ping running and did another prom capture on the VLAN interface (192.168.20.1) and there is no traffic at all.

To make sure it is not the infrastructure in between I setup another OPNsense on my test HyperV where the two test devices are running on and just put them on vSwitch. The two devices has the vlan configured directly in the VM configuration. The opnsense has no VLANs on the VM configuration but on the web interface but with the same result. I cant reach the VLAN interfaces.

**Comment by overcore456** [comment]
OMG
I dont whats happend, but now it works and I have no idea why. :-( The ping just startet to work. I have the suspicion that the firewall my be not processed all settings immediately?

**Comment by overcore456** [comment]
I try to investigate further and my colleague seems to work on an OpenVPN Site2Site Tunnel between two sites and this seems to have an impact on the whole VLAN configuration. We try to figure our what the reason is but so far there is no hint or pattern.

**Comment by fichtner** [comment]
Hi Marco,

Whew, ok, that's progress. I don't believe the tunnel does have an impact on VLAN itself, but maybe the two of you were configuring the firewall at the same time, leaving it with an inconsistent state (would be an OPNsense problem, not a user problem)?


Cheers,
Franco

**Comment by overcore456** [comment]
Hi Franco,

Ok, but we just did this today... it actually didn't work a couple of days ago where not worked together on this firewall. How can we make sure the firewall is consistent right now? Currently just one VLAN is working and the other one not. I can understand this at all. :-(

Cheers

**Comment by fichtner** [comment]
Hi Marco,

Did this resolve in any way (even if moved away from OPNsense)?


Cheers,
Franco
